File: 3761/Main.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int reverseNumber(int n) {
        int revNum = 0;

        while (n > 0) {
            revNum = (revNum * 10) + (n % 10);
            n /= 10;
        }

        return revNum;
    }

    int minMirrorPairDistance(vector<int>& nums) {
        map<int, int> indx;

        int n = nums.size();
        int answer = n;

        for (int i = 0; i < n; i++) {

            if (indx[nums[i]]) {
                answer = min(answer, i - indx[nums[i]] + 1);
            }
            indx[reverseNumber(nums[i])] = i + 1;
        }

        return answer == n ? -1 : answer;
    }
};

File: 3761/Main.cs
using System;
using System.Collections.Generic;

class Solution {
    public int ReverseNumber(int n) {
        int revNum = 0;
        while (n > 0) {
            revNum = revNum * 10 + (n % 10);
            n /= 10;
        }
        return revNum;
    }
    
    public int MinMirrorPairDistance(int[] nums) {
        Dictionary<int, int> indx = new Dictionary<int, int>();
        int n = nums.Length;
        int answer = n;
        
        for (int i = 0; i < n; i++) {
            if (indx.ContainsKey(nums[i])) {
                answer = Math.Min(answer, i - indx[nums[i]] + 1);
            }
            indx[ReverseNumber(nums[i])] = i + 1;
        }
        
        return answer == n ? -1 : answer;
    }
}